moveit_core/trajectory_processing/src/iterative_torque_limit_parameterization.cpp

```diff
@@ -90,6 +90,87 @@ bool IterativeTorqueLimitParameterization::computeTimeStampsWithTorqueLimits(
     return false;
   }
 
+  /* Create a mutable copy of `velocity_limits` and `acceleration_limits`;
+   * fill them in from the robot description if they are empty.
+   * TODO(cj): Factor out limit resolution; common/duplicated in TOTG as well.
+   */
+  std::unordered_map<std::string, double> mutable_vel_limits = velocity_limits;
+  std::unordered_map<std::string, double> mutable_accel_limits = acceleration_limits;
+
+  // Lambda for validating limits
+  auto validate_limit = [](const char* type, double value, const std::string& name) {
+    if (value <= std::numeric_limits<double>::epsilon())
+    {
+      ROS_ERROR_NAMED(LOGNAME, "Invalid %s limit %f for joint '%s'. Must be greater than zero!", type, value,
+                      name.c_str());
+      return false;
+    }
+    return true;
+  };
+
+  const std::vector<std::string>& joint_names = group->getActiveJointModelNames();
+  size_t num_joints = joint_names.size();
+  const robot_model::JointBoundsVector joint_bounds = group->getActiveJointModelsBounds();
+
+  for (size_t j = 0; j < num_joints; ++j)
+  {
+    const std::string& name = joint_names[j];
+    // Each element in joint_bounds is a pointer to a Bounds (i.e., vector<VariableBounds>)
+    const robot_model::JointModel::Bounds* bounds_ptr = joint_bounds[j];
+    if (bounds_ptr->size() != 1)
+    {
+      ROS_ERROR_NAMED(LOGNAME, "Cannot handle bounds for multi-variable joint '%s'!", name.c_str());
+      return false;
+    }
+    const moveit::core::VariableBounds vb = (*bounds_ptr)[0];
+
+    // Resolve velocity limit: Fill in if not already provided.
+    auto it = mutable_vel_limits.find(name);
+    if (it == mutable_vel_limits.end())
+    {
+      if (vb.velocity_bounded_)
+      {
+        double limit = std::min(std::fabs(vb.max_velocity_), std::fabs(vb.min_velocity_));
+        if (!validate_limit("velocity", limit, name))
+          return false;
+        mutable_vel_limits[name] = limit;
+      }
```

Comment:
seems weird to use the `min` here. shouldn't we use the max?

Reply:
same for accel

Reply:
Remembered why I did this now -- two reasons:
1. This is how other time parameterization algos resolve limits. They do this because (2)
2. `min_{velocity,acceleration}_` is just set to `-max_{velocity,acceleration}_`. Really the underlying assumption that's being taking advantage of is that these are _signed_ limits, not absolute; we want the most restrictive limit here, so we take the min of the magnitudes of the signed limits. The default value is 0, which we'll only get if a limit isn't set, and the validator will catch this and skip the parameterization since limits weren't defined.

Given that, I'm gonna leave these as is.
![image](https://github.com/user-attachments/assets/be86019a-60ce-437c-8a7d-95d424f397be)